custom/icds_reports/tasks.py

```diff
@@ -1745,20 +1745,29 @@ def get_data_not_in_ucr(status_record):
         doc_ids = [val[0] for val in chunk]
         docs_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
         for doc_id, doc_subtype, sql_modified_on in chunk:
-            if doc_id not in docs_in_ucr:
+            in_ucr = False
+            for ucr_doc in docs_in_ucr:
+                if ucr_doc[0] == doc_id:
+                    in_ucr = True
+                    # For the cases which are outdated.  This condition also handles the time drift of 1 sec
+                    # between main db and ucr db. i.e  doc will even be included when inserted_at-sql_modified_on <= 1 sec
+                    if sql_modified_on - ucr_doc[1] >= timedelta(seconds=-1):
```

Comment:
Added this `if` condition based on my best understanding of the time drifts between the database. Refering (https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/management/commands/stale_data_in_ucr.py#L102) which may be wrongly handling it